lib/tpm2_util.c

```diff
@@ -345,3 +352,40 @@ void tpm2_util_public_to_yaml(TPM2B_PUBLIC *public) {
         tpm2_tool_output("\n");
     }
 }
+
+tpm2_context_object_type tpm2_util_string_extract_object_name_type(const char *objectstr,
```

Comment:
This should probably load up an object if it's not a handle. Maybe give it a struct like so:
```c
struct loaded_object {
    HANDLE handle;
    const char *path;
}
```
Make a complimentary routine:
```c
tpm2_object_save(loaded_object *o) {
    if ( o->path) {
        // save context
    }
}
```
That's just a suggestion, im not 100% on it.